Libs/MRML/Core/vtkMRMLBitStreamVolumeNode.cxx

```diff
@@ -0,0 +1,235 @@
+#include "vtkMRMLBitStreamVolumeNode.h"
+
+// OpenIGTLink includes
+//#include "igtlioVideoDevice.h"
+
+// MRML includes
+#include "vtkMRMLScene.h"
+#include "vtkMRMLNRRDStorageNode.h"
+
+// VTK includes
+#include <vtkCollection.h>
+#include <vtkMatrix4x4.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkXMLUtilities.h>
+
+//----------------------------------------------------------------------------
+vtkMRMLNodeNewMacro(vtkMRMLBitStreamVolumeNode);
+
+//----------------------------------------------------------------------------
+// vtkMRMLBitStreamVolumeNode methods
+
+//-----------------------------------------------------------------------------
+vtkMRMLBitStreamVolumeNode::vtkMRMLBitStreamVolumeNode()
+{
+  this->compressionDevice = NULL;
+  IsCopied = false;
+  KeyFrameReceived = false;
+  KeyFrameDecoded = false;
+  KeyFrameUpdated = false;
+  MessageBufferValid = false;
+  KeyFrameBuffer = "";
+  MessageBuffer = "";
+}
+
+//-----------------------------------------------------------------------------
+vtkMRMLBitStreamVolumeNode::~vtkMRMLBitStreamVolumeNode()
+{
+  if (compressionDevice != NULL)
+     compressionDevice->Delete();
+}
+
+//---------------------------------------------------------------------------
+int vtkMRMLBitStreamVolumeNode::ObserveOutsideCompressionDevice(vtkMRMLCompressionDeviceNode* device)
+{
+  if (device)
+  {
+    device->AddObserver(device->GetDeviceContentModifiedEvent(), this, &vtkMRMLBitStreamVolumeNode::ProcessDeviceModifiedEvents);
+    //------
+    std::string bitStream = std::string(device->GetContent()->frameMessage);
```

Comment:
What if `device->GetContent()->frameMessage` is null ? I anticipate it will seg fault ..

Reply:
thanks for pointing out ,  device->GetContent() might return NULL. 
if device->GetContent() is not NULL. 
device->GetContent()->frameMessage won't be NULL, as frameMessage is of type string.